code/game/machinery/doors/airlock.dm

```diff
@@ -999,9 +1007,10 @@
 		var/wgw = askingPDA.msg_input(user, "Enter an access request message for the owner of this airlock:")
 		if(!wgw)//blank entry cancels
 			return ..()
-		//TODO wgw = "This is an automated airlock access request. Message: \"[wgw]\" Click here to open airlock (<--magic string that runs a proc to check access/open the airlock when clicked in the text chat)"
-		askingPDA.send_message(user, finalPDA, FALSE, wgw)
-		askingPDA.door_request_cd = world.time + 100//TODO or whatever the PDA cooldown is
+		wgw = "This is an automated airlock access request. Message: \"[wgw]\" <a href='?src=[REF(src)];remoteOpen=TRUE'>Click here to open airlock</a>"
+		if(askingPDA.send_message(user, finalPDA, FALSE, wgw))
```

Comment:
code seems to get this far before the runtime happens. the Head of Personnel never actually receives the PDA message

Reply:
looks like `finalPDA` is `null` because you don't initialize it to `list()`, so `finalPDA += HoP_PDA` just overwrites the null with `HoP_PDA` instead of adding it to a list because that's how `null +=` works I guess

Reply:
That's how BYOND works